x/gov/keeper/deposit.go

```diff
@@ -162,6 +163,66 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	return activatedVotingPeriod, nil
 }
 
+// BurnAndSendDepositsToCommunityPool will burn the (deposits * proposal_cancel_burn_rate) amount of proposal deposits
+// and send remaining deposits of the proposal to community pool.
+func (keeper Keeper) BurnAndSendDepositsToCommunityPool(ctx sdk.Context, proposalID uint64, destAddress string, totalDeposits []sdk.Coin) error {
+	store := ctx.KVStore(keeper.storeKey)
+
+	proposalCancelRate := keeper.GetParams(ctx).ProposalCancelRatio
+	// burn the deposits * proposal_cancel_rate amount from proposal deposits (gov module)
+	burnRate := sdk.MustNewDecFromStr(proposalCancelRate)
+
+	var burnDepositAmount sdk.Coins
+
+	if burnRate.IsPositive() {
+		for _, deposit := range totalDeposits {
+			burnAmount := sdk.NewCoin(
+				deposit.Denom,
+				sdk.NewDecFromInt(deposit.Amount).Mul(burnRate).RoundInt(),
+			)
+			burnDepositAmount = burnDepositAmount.Add(burnAmount)
+		}
+	}
+
+	// burn the deposits
+	if !burnDepositAmount.IsZero() {
+		err := keeper.bankKeeper.BurnCoins(ctx, types.ModuleName, burnDepositAmount)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
So this is wrong:
1. if `fee>0 && destAddress == ""` then we burn the fee
2. if `fee>0 && destAddress != ""` then we send the fee to `destAddress`
3. if `reminder != 0` we send the reminder back to the depositors.

Reply:
Done